### PR TITLE
Images in high resolution for SEO in SSR

### PR DESCRIPTION
**Pull request by mharsat**

### CODE REVIEW

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 1]
```diff
@@ -47,33 +54,31 @@ class ImageViewer extends React.Component {
     if (this.props.dataUrl) {
       imageUrl.preload = imageUrl.highres = this.props.dataUrl;
     } else {
+      let requiredWidth, requiredHeight;
       imageUrl.preload = getImageSrc(src, helpers);
-      if (this.state.container) {
+      if (isSeo) {
+        requiredWidth = src && src.width <= SEO_IMAGE_WIDTH ? src.width : SEO_IMAGE_WIDTH;
```
Comment (lxgreen) [inline packages/plugin-image/web/src/image-viewer.jsx]:
```suggestion
        requiredWidth = src && src.width && Math.min(src.width, SEO_IMAGE_WIDTH);
```

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 2]
```diff
@@ -32,8 +33,14 @@ class ImageViewer extends React.Component {
     }
   }
 
+  calculateHeight(width = 1, src) {
```
Comment (lxgreen) [inline packages/plugin-image/web/src/image-viewer.jsx]: 👍

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 3]
```diff
@@ -47,33 +54,31 @@ class ImageViewer extends React.Component {
...
+      } else if (this.state.container) {
         const { width } = this.state.container.getBoundingClientRect();
-        let requiredWidth = width || src.width || 1;
+        requiredWidth = width || src.width || 1;
         if (this.context.isMobile) {
-          const isSSR = typeof window === 'undefined';
           //adjust the image width to viewport scaling and device pixel ratio
-          requiredWidth *= (!isSSR && window.devicePixelRatio) || 1;
-          requiredWidth *= (!isSSR && window.screen.width / document.body.clientWidth) || 1;
+          requiredWidth *= (!isSSR() && window.devicePixelRatio) || 1;
```
Comment (lxgreen) [inline packages/plugin-image/web/src/image-viewer.jsx]: 👍

File: packages/editor/web/src/RichContentEditor/RichContentEditor.jsx [hunk 4]
```diff
@@ -53,7 +53,17 @@ class RichContentEditor extends Component {
...
+    const {
+      theme,
+      t,
+      locale,
+      anchorTarget,
+      relValue,
+      helpers,
+      config,
+      isMobile,
+      isSeo,
```
Comment (lxgreen) [inline packages/editor/web/src/RichContentEditor/RichContentEditor.jsx]: I'd prefer more meaningful name

Comment (mharsat) [reply packages/editor/web/src/RichContentEditor/RichContentEditor.jsx]: changed to `shouldRenderOptimizedImages`

File: docs/RichContentEditorApi.md [hunk 5]
```diff
@@ -24,3 +24,7 @@ The return value is a function that is used as the `draft-js` [handleReturn canc
 ## `config`
 
 The `config` prop allow to customize various aspects of the `RichContentEditor`, such as [toolbars](./ToolbarCustomization.md), [common UI settings](./UiSettings.md), and [plugins](./PluginCustomization.md).
+
+## `shouldRenderOptimizedImages`
```
Comment (lxgreen) [inline docs/RichContentEditorApi.md]: 👍

**Comment by mharsat** [comment]
fixes #320

**Pull request merged**